doc/source/area-detector.rst

```diff
@@ -269,6 +269,168 @@ The vendor specific details are embedded in the cams
+
+    2. Component (Cpt) can be replaced with ADComponent(ADCpt) if this is a
+    builtin areadetector component as this includes the possibility of using
+    the Areadetector documentation.
+
+    3. PluginClass can be PluginBase, CamBase or any plugin/cam class that
+    inherits from either of these.
+
+
+
+The Areadetector device should then be instantiated using:
+
+.. code-block:: python
+
+    ADdevice_name = Some_Areadetector_Device_Class(Areadetector_device_PV_suffix,
+                                                  name = 'ADdevice_name')
+
+
+Custom Plugins or Cameras
+=========================
+For custom hardware based on area-detector it may be necesary to add a custom
+plugin or camera class, this section will cover what is required. Both
+'plugins' and 'cameras' act in the same way, but have slightly different 'base'
+attributes, hence they have different 'base classes'. New Plugin classes should
+inherit from ``ophyd.areadetector.base.PluginBase`` while new Camera classes
+should inherit from ``ophyd.areadetector.cam.CamBase``. Both should have the
+following PV structure (replace 'plugin' with 'cam' for cameras):
+
+.. code-block:: python
+
+    PV = Areadetector_device_PV_prefix:Plugin_suffix:attribute_suffix
+
+As an example, for the 'max value' component of the built-in areadetector
+'stats' class this looks like:
+
+.. code-block:: python
+
+    Areadetector_device_PV_suffix:Stats:max_value
+
+where ``Areadetector_device_PV_suffix`` is the PV name for the Area detector
+device, ``plugin_suffix = Stats`` is the 'stats' Plugin suffix and
+``attribute_suffix = max_value`` is the 'max value' attribute suffix.
+
+
+In order to create the class then the following code is required (where ``XXX``
+is the name of the plugin:
+
+.. code-block:: python
+
+    from ophyd.areadetector.base import ADComponent as ADCpt, ad_group,
+                                        EpicsSignalWithRBV
+    from iohyd.signal import EpicsSignal, EpicsSignalRO
+    from ophyd.device import DynamicDeviceComponent as DDCpt
+    from ophyd.areadetector.plugins import PluginBase
+
+    Class XXXplugin(PluginBase', optional filestore_mixin'):
+        '''An areadetector plugin class that does ......'''
+        _suffix_re = '"Plugin_suffix"\d:'
+
+        # ADD ATTRIBUTES AS COMPONENTS HERE USING THE SYNTAX
+        attribute_name = Cpt(Type, attribute_suffix)
+            # where 'Type' is EpicsSignal, EpicsSignalRO, EpicsSignalWithRBV,..
+
+        # ADD ATTRIBUTE GROUPS AS COMPONENTS USING THE SYNTAX
+        group_name = DDCpt(ad_group(Type,
+                                    (attribute_1_name, attribute_1_suffix),
+                                    (attribute_2_name, attribute_2_suffix),
+                                    ...................
+                                    (attribute_n_name, attribute_n_suffix))
+
+.. note::
+
+    1. PluginBase can be swapped out for CamBase, FilePlugin or any other
```

Comment:
These should be links, using the `:class:` syntax mentioned about so people can click to learn more.

Reply:
have made changes at all places